backend/app/chat/routers.py

```diff
@@ -17,7 +17,7 @@
     NewChatRequest,
 )
 
-router = APIRouter(dependencies=[Depends(authenticate_key)], tags=["Chat endpoints"])
+router = APIRouter(dependencies=[Depends(get_current_user)], tags=["Chat endpoints"])
```

Comment:
We can now also get userid from the token and not require it in the `/chat` request schema

Reply:
ideally, we also have a test case where we access the endpoints with username/password auth. And with both user and admin roles.

Reply:
@sidravi1 Can we implement the `user_id` switcherooo in a later PR? I fear this one is getting quite large.